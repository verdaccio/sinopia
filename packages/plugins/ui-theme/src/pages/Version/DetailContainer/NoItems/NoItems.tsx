import React from 'react';
import Text from 'verdaccio-ui/components/Text';

interface Props {
  text: string;
  className?: string;
}

const NoItems: React.FC<Props> = ({ className, text, ...props }) => (
  // eslint-disable-next-line verdaccio/jsx-spread
  <Text {...props} className={className} gutterBottom={true} variant="subtitle1">
    {text}
  </Text>
);

export default NoItems;
