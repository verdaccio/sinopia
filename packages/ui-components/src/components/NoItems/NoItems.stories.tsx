import React from 'react';

import { default as NoItems } from '.';

export default {
  title: 'Components/Detail/No Items',
};

export const NoItemsText: any = () => {
  return <NoItems text="no items" />;
};
