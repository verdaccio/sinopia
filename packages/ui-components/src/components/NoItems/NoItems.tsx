import Typography from '@mui/material/Typography';
import React from 'react';

interface Props {
  text: string;
  className?: string;
}

const NoItems: React.FC<Props> = ({ className, text, ...props }) => (
  // eslint-disable-next-line verdaccio/jsx-spread
  <Typography {...props} className={className} gutterBottom={true} variant="subtitle1">
    {text}
  </Typography>
);

export default NoItems;
