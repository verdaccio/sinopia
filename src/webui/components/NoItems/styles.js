/**
 * @prettier
 * @flow
 */

import styled from 'react-emotion';

export const Wrapper = styled.div`
  && {
    margin: 5em 0;
  }
`;
