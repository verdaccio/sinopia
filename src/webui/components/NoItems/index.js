/**
 * @prettier
 * @flow
 */

import React from 'react';
import { IProps } from './types';
import { Wrapper } from './styles';

const NoItems = ({ text }: IProps) => (
  <Wrapper>
    <h2>{text}</h2>
  </Wrapper>
);

export default NoItems;
