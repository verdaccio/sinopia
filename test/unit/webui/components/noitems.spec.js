/**
 * NoItems component
 */

import React from 'react';
import { shallow, mount } from 'enzyme';
import NoItems from '../../../../src/webui/components/NoItems/index';

console.error = jest.fn();

describe('<NoItem /> component', () => {
  it('should load the component in default state', () => {
    const wrapper = mount(<NoItems />);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should set html from props', () => {
    const props = {
      text: 'This is a test string'
    };
    const wrapper = shallow(<NoItems {...props} />);
    expect(wrapper.find('h2').text()).toEqual('This is a test string');
    expect(wrapper.html()).toMatchSnapshot();
  });
});
