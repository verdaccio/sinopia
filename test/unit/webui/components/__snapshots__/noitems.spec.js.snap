// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NoItem /> component should load the component in default state 1`] = `"<div class=\\"css-6ghnff e1ggl3im0\\"><h2></h2></div>"`;

exports[`<NoItem /> component should set html from props 1`] = `"<div class=\\"css-6ghnff e1ggl3im0\\"><h2>This is a test string</h2></div>"`;
